client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +159,66 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+// @see https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/delete/
+export const deleteComment = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, commentId } = action;
+	const comment = getSiteComment( getState(), siteId, commentId );
+
+	dispatch( removeComment( siteId, postId, commentId ) );
+
+	dispatch(
+		http(
```

Comment:
`local()` should probably never be used outside of the data layer and I think it's very reasonable to check for the placeholder here